sui/src/benchmark/transaction_creator.rs

```diff
@@ -248,10 +176,89 @@ impl TransactionCreator {
             &validator_preparer.committee,
         );
 
-        self.object_id_offset += chunk_size * num_chunks;
-
         validator_preparer.update_objects_for_validator(txn_objects, address);
 
         signed_txns
     }
+
+    fn make_gas_objects(
+        &mut self,
+        address: SuiAddress,
+        tx_count: usize,
+        batch_size: usize,
+        obj_id_offset: ObjectID,
+    ) -> Vec<(Vec<Object>, Object)> {
+        let total_count = tx_count * batch_size;
+        let mut objects = vec![];
+        let mut gas_objects = vec![];
+        // Objects to be transferred
+        ObjectID::in_range(obj_id_offset, total_count as u64)
+            .unwrap()
+            .iter()
+            .for_each(|q| objects.push(create_gas_object(*q, address)));
+
+        // Objects for payment
+        let next_offset = objects[objects.len() - 1].id();
+
+        ObjectID::in_range(next_offset.next_increment().unwrap(), tx_count as u64)
```

Comment:
Could we not just create random object IDs? Or is the issue that they need to exist over all authorities so we need a deterministic generation process? This arithmetic on object IDs is very error prone and misleading.

Reply:
The reason I have the range arithmetic is so that load gen TXes do not clash. I want them to operate in specific object id ranges. This makes debugging much easier